aws/resource_aws_db_parameter_group.go

```diff
@@ -295,6 +295,26 @@ func resourceAwsDbParameterGroupUpdate(d *schema.ResourceData, meta interface{})
 			}
 			d.SetPartial("parameter")
 		}
+
+		// Reset parameters that have been removed
+		parameters, err = expandParameters(os.Difference(ns).List())
+		if err != nil {
+			return err
+		}
+		if len(parameters) > 0 {
+			parameterGroupName := d.Get("name").(string)
+			resetOpts := rds.ResetDBParameterGroupInput{
+				DBParameterGroupName: aws.String(parameterGroupName),
+				Parameters:           parameters,
```

Comment:
It looks like by default, all parameters are reset so we should add this in to prevent that:

```suggestion
				Parameters:           parameters,
				ResetAllParameters:   aws.Bool(false),
```